Assets/Scripts/Model/Weapon.cs

```diff
@@ -98,7 +105,7 @@ public CanReceiveDamage getAvailableTarget()
             var target = targets[0];
 
             // Check if target is already dead
-            if (target.Equals(null))
+			if (target.Equals(null) || target == null)
```

Comment:
If you had missing references, the order should be switched, since the expression that gives you the error comes first even after you changed xd